BootloaderCommonPkg/Library/BootloaderCommonLib/BootloaderCommonLib.c

```diff
@@ -38,7 +38,8 @@ CpuHalt (
   DEBUG ((DEBUG_ERROR, "\nSTAGE_%a: System halted!\n", mStage[GetLoaderStage()]));
 
   // Flush all console buffer if serial console is not active
-  if ((PcdGet32 (PcdDebugOutputDeviceMask) & DEBUG_OUTPUT_DEVICE_SERIAL_PORT) == 0) {
+  if ( ((PcdGet32 (PcdDebugInterfaceFlags) & (DEBUG_DEVICE_ISA_SERIAL | DEBUG_DEVICE_SERIALIO)) == 0) &&
+       ((PcdGet32 (PcdDebugInterfaceFlags) & DEBUG_DEVICE_RAM) != 0)) {
```

Comment:
Why do we have dependency on DEBUG_DEVICE_RAM ?

Reply:
DEBUG_DEVICE_RAM means the debug info is saved in memory. If this one is not enabled, the debug info should be empty. no point to do output.

Reply:
This condition is handled already below.  If it is not enabled, the UsedLength will be the same as Header Length.   So nothing will be output from the code.